modules/container/set.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package container
+
+type Set[T comparable] map[T]struct{}
+
+// NewSet creates a set and adds the specified elements to it.
+func NewSet[T comparable](values ...T) Set[T] {
```

Comment:
How about name it `MakeSet`

Reply:
Or `SetOf`? Or `SetOfValues`?

Reply:
> Or `SetOf`? Or `SetOfValues`?

I'm also OK for the name. I just want to avoid to use `NewXXX` because commonly it will return a pointer.